database/src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -91,33 +93,50 @@ public boolean deleteTable(Class<? extends BaseEntity> entityClass) {
 
     @Override
     public <T extends BaseEntity> T addNewRecordToTable(T entity) {
-        if (entity.getId() != null) {
-            throw new IdProvidedManuallyException(ID_PROVIDED_MANUALLY);
-        }
+        validateIdNotProvidedManually(entity);
 
         Class<? extends BaseEntity> entityClass = entity.getClass();
         Path databasePath = Path.of(getDatabasePath(entityClass));
-
-        if (!Files.exists(databasePath)) {
-            LOG.error(DB_FILE_NOT_EXIST + ": {}", databasePath.toAbsolutePath());
-            throw new DatabaseDoesNotExistException(DB_FILE_NOT_EXIST);
-        }
+        verifyDatabaseExists(databasePath);
 
         List<T> entities = deserializeEntities(entityClass, readDatabaseFile(databasePath));
-
         String entityClassName = entityClass.getName();
 
-        entity.setId(entityIds.merge(entityClassName, 1, Integer::sum));
+        assignEntityId(entity, entityClassName);
         entities.add(entity);
 
         saveEntitiesToDatabase(entities, databasePath);
         return entity;
     }
 
+    @Override
+    public <T extends BaseEntity> Iterable<T> addNewRecordsToTable(Class<? extends BaseEntity> entityClass,
+                                                                   List<T> entities) {
+        if (entities == null || entities.isEmpty()) {
+            throw new IllegalArgumentException(ENTITIES_LIST_NULL_OR_EMPTY);
```

Comment:
Please use custom exception which extends from `BadRequestException` to return 400 error code instead of 500.